js/lib/baseDomain.js

```diff
@@ -9,14 +9,27 @@ const LRUCache = require('lru_cache/core').LRUCache
 
 let cachedBaseDomain = new LRUCache(50)
 
+const checkASCII = function (str) {
+  if (typeof str !== 'string') {
+    return false
+  }
+  for (let i = 0; i < str.length; i++) {
+    if (str.charCodeAt(i) > 127) {
+      return false
+    }
+  }
+  return true
+}
+
 /**
  * Returns base domain for specified host based on Public Suffix List.
  * @param {string} hostname The name of the host to get the base domain for
  */
 
 module.exports.getBaseDomain = function (hostname) {
   // decode punycode if exists
-  if (hostname.indexOf('xn--') >= 0) {
+  if (hostname.indexOf('xn--') >= 0 &&
+    checkASCII(hostname)) {
```

Comment:
i figured out why this was crashing. the `hostname` that gets checked here when i go to http://xn--jp%29-5y3b.f1-live.com/f1/jp/headlines/news/detail/060912193351.shtml is `xn--jp`, which is valid ASCII but invalid punycode. so punycode.js throws a (different) error, and the crash happens. I verified that wrapping `hostname.toUnicode` in a try-catch fixes it for me.